From 9a830f6c7a2c3daa5fa3df6853b315b84741af91 Mon Sep 17 00:00:00 2001
From: Denis Oliver Kropp <dok@shizo.zion.home>
Date: Wed, 30 May 2007 16:44:34 +0200
Subject: [PATCH] Start using architecture dependent SHM virtual address area. User space still uses 0x20000000 for a few root pages though.

For x86_64 (address space!) use 0x500000010000 with length 0x0FFFFFFEF000.

For mips use 0x50010000 and others 0x20010000 as before, length 0x1FFEF000.

If there are conflicts on other platforms, please add an elsif :)
---
 drivers/char/fusion/shmpool.h |   12 ++++++++++++
 1 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/drivers/char/fusion/shmpool.h b/drivers/char/fusion/shmpool.h
index 06c6c38..0bbb588 100644
--- a/drivers/char/fusion/shmpool.h
+++ b/drivers/char/fusion/shmpool.h
@@ -18,8 +18,20 @@
 #include "fusiondev.h"
 #include "types.h"
 
+#ifdef __x86_64__
+#define FUSION_SHM_BASE    0x500000010000     /* virtual base address */
+#define FUSION_SHM_SIZE    0x0FFFFFFEF000     /* size of virtual address space */
+#else
+
+
+#ifdef __mips__
+#define FUSION_SHM_BASE    0x50010000     /* virtual base address */
+#else
 #define FUSION_SHM_BASE    0x20010000     /* virtual base address */
+#endif
+
 #define FUSION_SHM_SIZE    0x1FFEF000     /* size of virtual address space */
+#endif
 
 /* module init/cleanup */
 
-- 
1.7.3.3

